aion/web/controllers/question_controller.ex

```diff
@@ -2,43 +2,24 @@ defmodule Aion.QuestionController do
   use Aion.Web, :controller
 
   alias Aion.Question
-  alias Aion.Answer
-  alias Aion.Subject
+  alias Aion.QuestionTransactions
 
   def index(conn, _params) do
     questions = Repo.all(Question)
     render(conn, "index.json", questions: questions)
   end
 
-  def create(conn, %{"question" => question_params, "answers" => answers_content, "subject" => subject_id}) do
-    question_subject = Repo.get(Subject, subject_id)
-
-    transaction_result = Repo.transaction(fn ->
-      question_changeset = Question.changeset(%Question{}, Map.put(question_params, "belongs_to", question_subject))
-      case Repo.insert(question_changeset) do
-        {:ok, question} ->
-          answer_changeset = Answer.changeset(%Answer{}, %{"content" => answers_content, "belongs_to" => question})
-          case Repo.insert(answer_changeset) do
-            {:ok, answer} ->
-              question
-            {:error, error_answer_changeset} ->
-              Repo.rollback(error_answer_changeset)
-          end
-        {:error, error_question_changeset} ->
-          Repo.rollback(error_question_changeset)
-      end
-    end)
-
-    case transaction_result do
-      {:ok, question} ->
+  def create(conn, %{"question" => question, "answers" => answers, "subject" => subject_id}) do
+    transaction = QuestionTransactions.create_question_with_answers(question, answers, subject_id)
+    case transaction |> Repo.transaction do
```

Comment:
Not sure if naming 'function to run' a 'transaction' is a fortunate choice since the function that is going to call it is named [**transaction**](https://hexdocs.pm/ecto/Ecto.Adapter.Transaction.html) itself. Gets a bit confusing. 
![image](https://user-images.githubusercontent.com/15965147/29290121-3da9e75a-813f-11e7-8352-70eaaf87c2e7.png)
Also I don't think it's really clear to use pipeline here. Imho since it's only one function call you could just say ```case Repo.transaction(transaction) do```

Anyway, I would bind the value of the operation to a variable like this:
```
transaction_result = Repo.transaction(transaction)
case transaction_result do
  ...
```